src/main/java/org/opensearch/ad/caching/CacheBuffer.java

```diff
@@ -324,21 +335,21 @@ public void maintenance() {
                     // put: not a problem as we are unlikely to maintain an entry that's not
                     // already in the cache
                     // remove method saves checkpoint as well
-                    remove(entityModelId);
-                } else {
-                    // we can have ConcurrentModificationException when serializing
-                    // and updating rcf model at the same time. To prevent this,
-                    // we need to have a deep copy of models or have a lock. Both
-                    // options are costly.
-                    // As we are gonna retry serializing either when the entity is
-                    // evicted out of cache or during the next maintenance period,
-                    // don't do anything when the exception happens.
-                    checkpointDao.write(modelState, entityModelId);
+                    removedStates.add(remove(entityModelId));
+                } else if (random.nextInt(6) == 0) {
```

Comment:
Why choose 6 ?  Add some comments?

Reply:
The comments/explanation are in the next few lines.

Reply:
oh, just see it. Sorry I jumped to a call when I reach this line. Continue reading this PR now.